### PR TITLE
auto case OCP-25365 OCP-18147

### PR DESCRIPTION
**Pull request by QiaolingTang**
@anpingli  PTAL, thanks.

Polarion cases: https://url.corp.redhat.com/5f6efcf

### CODE REVIEW

File: features/logging/4.x/collector.feature [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    Given evaluation of `pod` is stored in the :log_pod clipboard
+    Given I switch to cluster admin pseudo user
+    Given I use the "openshift-logging" project
+    Given evaluation of `cluster_logging('instance').collection_type` is stored in the :collection_type clipboard
+    And I wait up to 600 seconds for the steps to pass:
+    """
+    And I get the "project.<%= cb.proj.name %>" logging index information from a pod with labels "es-node-master=true"
+    Then the step should succeed
+    And the expression should be true> cb.index_data and cb.index_data.count > 0
+    """
```
Comment (akostadinov) [inline features/logging/4.x/collector.feature]: @pruan-rht , I see this check two times already. As if it waits for logging data to be present. Do we have a special step?

@QiaolingTang , cold you confirm purpose of this waiting?

Comment (QiaolingTang) [reply features/logging/4.x/collector.feature]: @akostadinov the ES may need some time to create the index and sometimes the collector needs time to collect the logs in projects, these condition always happen in a shared cluster due to limited resources, so I added the step to wait for the data to appear in the ES.

Comment (pruan-rht) [reply features/logging/4.x/collector.feature]: @QiaolingTang to be consistent, let's assume we are always do this check so it's better to wrap it in a step as suggested by @akostadinov

Comment (akostadinov) [reply features/logging/4.x/collector.feature]: @pruan-rht , so step like `Given log entries appear for the project`?

Comment (QiaolingTang) [reply features/logging/4.x/collector.feature]: @pruan-rht @akostadinov seems we have a step `When /^I wait(?: (\d+) seconds)? for the #{QUOTED} index to appear in the ES pod(?: with labels #{QUOTED})?$/ do |seconds, index_name, pod_labels|`, I'll try this step.

Comment (akostadinov) [reply features/logging/4.x/collector.feature]: Awesome, good catch!

**Comment by akostadinov** [comment]
Conflict :/ , please resolve conflicts and will merge.

File: features/logging/4.x/collector.feature (hunk 1 above)
Comment (pruan-rht) [reply features/logging/4.x/collector.feature]: @QiaolingTang yeah, that was written a while back.

**Comment by pruan-rht** [comment]
@QiaolingTang can you please resolve the conflicts?  Thx!

**Comment by QiaolingTang** [comment]
Conflicts are resolved, thank you all.

**Pull request merged**